ts/client/src/client.ts

```diff
@@ -1696,6 +1698,81 @@ export class MangoClient {
     return await this.sendAndConfirmTransactionForGroup(group, [ix]);
   }
 
+  public async serum3SettleFundsV2Ix(
+    group: Group,
+    mangoAccount: MangoAccount,
+    externalMarketPk: PublicKey,
+  ): Promise<TransactionInstruction> {
+    const serum3Market = group.serum3MarketsMapByExternal.get(
+      externalMarketPk.toBase58(),
+    )!;
+    const serum3MarketExternal = group.serum3ExternalMarketsMap.get(
+      externalMarketPk.toBase58(),
+    )!;
+
+    const [serum3MarketExternalVaultSigner, openOrderPublicKey] =
+      await Promise.all([
+        generateSerum3MarketExternalVaultSignerAddress(
+          this.cluster,
+          serum3Market,
+          serum3MarketExternal,
+        ),
+        serum3Market.findOoPda(this.program.programId, mangoAccount.publicKey),
+      ]);
+
+    const ix = await this.program.methods
+      .serum3SettleFundsV2(this.openbookFeesToDao)
+      .accounts({
+        v1: {
+          group: group.publicKey,
+          account: mangoAccount.publicKey,
+          owner: (this.program.provider as AnchorProvider).wallet.publicKey,
+          openOrders: openOrderPublicKey,
+          serumMarket: serum3Market.publicKey,
+          serumProgram: OPENBOOK_PROGRAM_ID[this.cluster],
+          serumMarketExternal: serum3Market.serumMarketExternal,
+          marketBaseVault: serum3MarketExternal.decoded.baseVault,
+          marketQuoteVault: serum3MarketExternal.decoded.quoteVault,
+          marketVaultSigner: serum3MarketExternalVaultSigner,
+          quoteBank: group.getFirstBankByTokenIndex(
+            serum3Market.quoteTokenIndex,
+          ).publicKey,
+          quoteVault: group.getFirstBankByTokenIndex(
+            serum3Market.quoteTokenIndex,
+          ).vault,
+          baseBank: group.getFirstBankByTokenIndex(serum3Market.baseTokenIndex)
+            .publicKey,
+          baseVault: group.getFirstBankByTokenIndex(serum3Market.baseTokenIndex)
+            .vault,
+        },
+        v2: {
+          quoteOracle: group.getFirstBankByTokenIndex(
+            serum3Market.quoteTokenIndex,
+          ).oracle,
+          baseOracle: group.getFirstBankByTokenIndex(
+            serum3Market.baseTokenIndex,
+          ).oracle,
+        },
+      })
+      .instruction();
+
+    return ix;
+  }
+
+  public async serum3SettleFundsV2(
```

Comment:
I'm wondering how to migrate here. Proposal and thoughts:
- Switching to the new instruction everywhere would cause the client code to depend on the new program deployment for a while, I'd rather avoid that.
- Change serum3SettleFunds() and serum3SettleFundsIx() to use V2 instructions when openbookFeesToDao==false. That way user code works correctly without changes.
- Possibly introduce serum3SettleFundsV1Ix() that asserts if openbookFeesToDao is false?
- It'd be cool if the client could determine the deployed program verison somehow and use that to decide what behaviors are supported. Is it easy to fetch the "version" field from the on-chain program idl?

Reply:
Additionally, we may want to prefer the v2 instruction globally at some point, if we can give the user additional fee-buyback benefits now that the oracles are passed in.

Reply:
ok so I will

* remove the serum3SettleFundsV2Ix and serum3SettleFundsV2
* make serum3SettleFundsIx and serum3SettleFunds use v1 ix if flag is true, use v2 if flag is false
* yes clients can determine, but would need to fetch on chain idl, maybe can do this async, if we see idl with version where v2 ix is present, then always use v2 ix, wdyt?